packages/components/src/Menu/Menu.tsx

```diff
@@ -106,80 +121,76 @@ export function Menu({ activator, items }: MenuProps) {
-    position.horizontal === "left" && styles.left,
-    position.horizontal === "right" && styles.right,
-  );
-
-  const wrapperClasses = classnames(styles.wrapper, {
-    [styles.fullWidth]: fullWidth,
-  });
-
   return (
-    <div
-      className={wrapperClasses}
-      ref={wrapperRef}
-      onClick={handleParentClick}
-    >
+    <div className={wrapperClasses} onClick={handleParentClick}>
+      <span ref={shadowRef} className={styles.shadowRef} />
       {React.cloneElement(activator, {
         onClick: toggle(activator.props.onClick),
         id: buttonID,
         ariaControls: menuID,
         ariaExpanded: visible,
         ariaHaspopup: true,
       })}
-      <AnimatePresence>
-        {visible && (
-          <>
-            <motion.div
-              className={styles.overlay}
-              onClick={toggle()}
-              variants={variation}
-              initial="overlayStartStop"
-              animate="done"
-              exit="overlayStartStop"
-              transition={{
-                type: "tween",
-                duration: 0.15,
-              }}
-            />
-            <motion.div
-              className={menuClasses}
-              role="menu"
-              aria-labelledby={buttonID}
-              id={menuID}
-              onClick={hide}
-              variants={variation}
-              initial="startOrStop"
-              animate="done"
-              exit="startOrStop"
-              custom={position}
-              transition={{
-                type: "tween",
-                duration: 0.25,
-              }}
-            >
-              {items.map((item, key: number) => (
-                <div key={key} className={styles.section}>
-                  {item.header && <SectionHeader text={item.header} />}
-
-                  {item.actions.map((action, index) => (
-                    <Action
-                      sectionLabel={item.header}
-                      key={action.label}
-                      shouldFocus={key === 0 && index === 0}
-                      {...action}
-                    />
-                  ))}
-                </div>
-              ))}
-            </motion.div>
-          </>
-        )}
-      </AnimatePresence>
+      <MenuPortal>
+        <AnimatePresence>
+          {visible && (
+            <>
+              <motion.div
+                className={styles.overlay}
+                onClick={toggle()}
+                variants={variation}
+                initial="overlayStartStop"
+                animate="done"
+                exit="overlayStartStop"
+                transition={{
+                  type: "tween",
+                  duration: 0.15,
+                }}
+              />
+              <div
+                ref={setPopperElement}
+                className={styles.popperContainer}
+                {...positionAttributes}
+                {...focusAttribute}
+              >
+                {items.length > 0 && (
```

Comment:
since we have nothing that says you must provide a non empty array, now that I'm using the focus trapping on the menu - it looks for the first focusable element to call `.focus()` on

however, if `items` is an empty array, it blows up 😱 

I don't think it's valid to have an empty menu. the UI before this was a silly looking sliver of a menu. we should probably not allow that, but for now this at least makes tests that were passing in empty `items` pass - and it guards against rendering that sliver